core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplainRequest.java

```diff
@@ -84,6 +90,8 @@ public ActionRequestValidationException validate() {
             }
             if (this.primary == null) {
                 validationException = addValidationError("primary must be specified", validationException);
+            } else if (this.primary && this.currentNodeId != null) {
+                validationException = addValidationError("currentNodeId can only be specified for a replica shard", validationException);
```

Comment:
it seems like that's a user error to provide a node id in that case and they probably should be alerted about it instead of silently ignoring it (maybe they meant to do something else like specify a replica shard, so then they would be confused about the output).  WDYT?